### ISSUE TITLE
[Bug] Client not attempting to reconnect

### ISSUE THREAD

**Issue by fa-sharp** [aembke/fred.rs#315]
Fred version - 10.0.0
Redis version - 7.4.1
Platform - mac
Deployment type - centralized

**Describe the bug**
The client doesn't attempt re-connection based on the reconnect policy. It correctly logs that the connection closed unexpectedly, but then I don't see any attempt at re-connection. If I call `client.is_connected()` after this, it still returns `true`, so I think something is going wrong here. This seems to be a regression - it was working as expected in the previous version.

Note that I'm using an individual client here, not a pool - not sure if that's making a difference.

**To Reproduce**

1. Setup a reconnect policy, and also setup an unresponsive timeout.
2. Connect to a Redis server with fred.rs, then shutdown the Redis server (or disconnect from internet).
3. Re-connection isn't attempted.

**Logs**
```json
{"level":"DEBUG","message":"fred-MR0vuty0o4: Resetting connection to localhost:6379 with error: Error { details: \"Connection closed.\", kind: IO }","target":"fred::router::utils","time":1733350218743}
{"level":"WARN","message":"fred-MR0vuty0o4: Ending reader task from localhost:6379 due to Some(Error { details: \"Connection closed.\", kind: IO })","target":"fred::router::responses","time":1733350218743}
{"level":"DEBUG","message":"fred-MR0vuty0o4: Checking reconnect state. Has policy: true, Is intentionally disconnecting: false","target":"fred::modules::inner","time":1733350218743}
{"level":"DEBUG","message":"fred-MR0vuty0o4: Defer reconnection to Some(Server { host: \"localhost\", port: 6379 }) after Error { details: \"Connection closed.\", kind: IO }","target":"fred::router::utils","time":1733350218743}
```

**Comment by aembke** [comment]
Thanks @fa-sharp, this will be fixed in 10.0.1 later today (https://github.com/aembke/fred.rs/pull/316).

**Comment by aembke** [comment]
Should be fixed in 10.0.1 (https://crates.io/crates/fred/10.0.1)
